### PR TITLE
add concurrent batched routes script with diffing and progress indication

### PR DESCRIPTION
**Pull request by kevinkreiser**
this script will run a bunch of routes in parallel and forward their output to a file named by the line number they appear on in the input file. if another directory is provided the output of this run will be diff'd against the output in that directory. ill make a more comprehensive routes test for liechtenstein before merging this.

here's the usage summary, args in [] are optional:

    kkreiser@k:~/sandbox/github/valhalla/thor/scripts$ ./batch.sh 
    Usage: ./batch.sh pathTestArgsFile [diffDir] [concurrency] [outDir]
    Example: ./batch.sh paRoutes
    Example: ./batch.sh paRoutes 2015_06_13_14_45_03
    Example: ./batch.sh paRoutes 2015_06_13_14_45_03 8
    Example: ./batch.sh paRoutes 2015_06_13_14_45_03 8 mySpecialDir

    Note: each line in your pathTestArgsFile should look like this:

    -o 47.118589,9.543217 -d 47.166110,9.511975 -t auto conf/valhalla.json

### CODE REVIEW

**Comment by dgearhart** [comment]
Ok to get started 
We do not want to capture the output from pathtest (because any additional debug would cause diffs) but rather write specfific files from pathtest. (or capture specific lines from output) directions file,edge attributes file, etc
Also, the origin and destination will at least have to have street name, city, state so we know what we are looking at - maybe pipe delimited since some addresses have commas

File: scripts/batch.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+#run all of the paths, make sure to cut off the timestamps
+#from the log messages otherwise every line will be a diff
+echo -e "\x1b[32;1mWriting routes from ${INPUT} with a concurrency of ${CONCURRENCY} into ${OUTDIR}\x1b[0m"
+cat "${INPUT}" | parallel --progress -k -C ' ' -P "${CONCURRENCY}" "pathtest {} 2>&1 | sed -e 's/^[^\[]*\[//' &> ${OUTDIR}/{#}.txt"
+
+#if we need to run a diff
+if [ -d "${DIFF}" ]; then
+	echo -e "\x1b[32;1mDiffing the output of ${DIFF} with ${OUTDIR}:\x1b[0m"
+	diff "${DIFF}" "${OUTDIR}"
```
Comment (dgearhart) [inline scripts/batch.sh]: save diffs to a diff directory

**Comment by gknisely** [comment]
agree with duane and then ship

**Comment by dgearhart** [comment]
looks good

File: scripts/batch.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+rm -f "${TMP}"
+
+#if we need to run a diff
+if [ -d "${DIFF}" ]; then
+	if [[ "${DIFF}" == *$(basename ${INPUT})* ]]; then
+		echo -e "\x1b[32;1mDiffing the output of ${DIFF} with ${OUTDIR} to ${DIFF}_${OUTDIR}_diff:\x1b[0m"
+		mkdir -p "${DIFF%/}_${OUTDIR}_diff"
+		find ${DIFF}/*.txt -printf "%f\n" | parallel --progress -P "${CONCURRENCY}" "diff ${DIFF}/{} ${OUTDIR}/{} > ${DIFF%/}_${OUTDIR}_diff/{}"
+	else
+		echo -e "\x1b[31;1mFiled to diff ${DIFF} with ${OUTDIR} as it looks like they were generated from different input\x1b[0m"
```
Comment (dgearhart) [inline scripts/batch.sh]: typo "Failed"

**Pull request merged**